### ISSUE TITLE
WindowsPhone8 Development

### ISSUE THREAD

**Issue by jamesoliverband** [cureos/accord#2]
It says that the assembly is incompatible. What do I have to do to get it run?

Portable.Accord.NET.sln if I open this it can only Import the unit tests
Accord.NET.sln if I open this it can Import at least everything and I can build it but as described above the dll can't be used in a WP8 Project.

**Comment by cureos** [comment]
Hi James,

Do I understand correctly that when you open the _Portable.Accord.NET_ solution, all PCL libraries are marked incompatible? Which Visual Studio version and flavor are you using? At least VS 2012 is required, and you will need Professional or higher, Express versions cannot build PCL projects.

Apart from this, I realize that the instructions for creating your own Windows Phone 8 or Windows Store project are scarce (to say the least...) I will try to add clearer instructions for that part as soon as possible. But I should point out that if you are creating for example a Windows Phone 8 application, you will need to reference the <b>_Phone_.System</b> and <b>_Phone_.System.Drawing</b> projects/DLL:s from the _Portable AForge_ solution. 

I plan to produce NuGet packages for PCL _AForge_ and _Accord_ if I get permission from the original authors of the respective frameworks. With NuGet packages in place, it will be possible to incorporate the PCL DLL:s even when developing with Visual Studio Express.

Best regards,
Anders

**Comment by jamesoliverband** [comment]
Do I understand correctly that when you open the Portable.Accord.NET solution, all PCL libraries are marked incompatible?  - Yes

I am using VS Express for Windows Phone. So currently I am stuck on this.

**Comment by cureos** [comment]
To overcome the problem that PCL libraries cannot be built in the Express versions of Visual Studio, I have now provided prebuilt binaries for the Portable AForge and Accord projects.

**Comment by jamesoliverband** [comment]
Hi Anders, 

I feel a bit stupid. But I downloaded the latest version and I can’t find the “prebuilt binaries for the Portable AForge and Accord projects”.

What do I do wrong. Thanks for your support.

Best regards,
Jamie

**Comment by cureos** [comment]
No worries, Jamie :-)

The prebuilt binaries are not included in the Git repository, but they are available as separate releases on Github. You'll find the binaries for AForge [here](https://github.com/cureos/aforge/releases/tag/v2.2.5), and the binaries for Accord [here](https://github.com/cureos/accord/releases/tag/v2.12.0.1).

Please note that when you are developing for WP8, you will need to reference the _WriteableBitmapEx_ assembly included with the AForge binaries, since the official _WriteableBitmapEx_ assembly for WP is _not_ string-named/signed.

Hope this helps!

Best regards,
Anders

**Comment by jamesoliverband** [comment]
Thanks Anders,

I downloaded now the correct libraries and tried to reference the AForge library but I again got this message:

What to do next?

Best regards,
Jamie

**Comment by cureos** [comment]
What message are you getting? 

Another developer reported that the DLL:s in the zip archive were blocked, so he needed to unblock them before he could use them in his project. Could this be the issue for you?

Are you also referencing the WP8 specific binaries (Any CPU should suffice) of _AForge.System.dll_ and _AForge.System.Drawing.dll_?

**Comment by jamesoliverband** [comment]
Hi, 
i think the Picture of the message dissapeared because I wrote it via mail. So here again:
![accord](https://f.cloud.github.com/assets/6304856/1878650/b4a236d0-793e-11e3-836d-d0a4bcc988ef.PNG)

I can verify that I referenced all 3 libraries. I also changed the rights on the dlls but yet no effect.

**Comment by jamesoliverband** [comment]
Here is also info about my visual Studio Version:

![accord2](https://f.cloud.github.com/assets/6304856/1878725/616bbc10-793f-11e3-8f1a-6d79c5c5a4ed.PNG)

**Comment by cureos** [comment]
Strange. Do you get the error message when you try to check _Accord.MachineLearning.dll_, or do you receive the message when clicking the OK button?

If I remember correctly, I have built all assemblies using VS 2013 Pro. I could try to build them in VS 2012 (Pro) and forward those binaries to you to see if this would help you succeed. I am not able to do that right away, though.

**Comment by jamesoliverband** [comment]
I first click Browse and select the dlls (it is really the same for every dll). Then when I click OK I get the error message.

Ok, Thanks. I hope this helps.

**Comment by cureos** [comment]
@jamesoliverband Do you get the error message even when you try to select the WP8 specific binaries (WriteableBitmapEx, AForge.System, AForge.System.Drawing), or does the error message only show up when you try to select any of the PCL assemblies?

**Comment by jamesoliverband** [comment]
Yes, I get that message with WriteableBitmapEx, AForge.System, AForge.System.Drawing, as well as with the pcl assemblies.

**Comment by jamesoliverband** [comment]
Hi Anders,

I now solved the misery. Instead of unblocking I was removing the write protection from the *.dll files. I tried out with the files you provided earlier all worked.

For all who run into the same issue. A picture says more than thousand words: 
![accord](https://f.cloud.github.com/assets/6304856/1883481/c5427e3a-7986-11e3-911c-35824bd43f1c.PNG)

Thanks for putting all the effort into my issue. 

Best regards,
Jamie Oliver
